relay-config/src/config.rs

```diff
@@ -726,31 +726,66 @@ impl Default for Http {
     }
 }
 
-fn buffer_max_connections() -> u32 {
-    30
+/// Default for max memory size, 500 MB.
+fn spool_envelopes_max_memory_size() -> usize {
+    524288000
 }
 
-fn buffer_min_connections() -> u32 {
+/// Default for max disk size, 500 MB.
+fn spool_envelopes_max_disk_size() -> usize {
+    524288000
+}
+
+/// Default for min connections to keep open in the pool.
+fn spool_envelopes_min_connections() -> u32 {
     10
 }
 
-/// Controls internal caching behavior.
+/// Default for max connections to keep open in the pool.
+fn spool_envelopes_max_connections() -> u32 {
+    20
+}
+
+/// Persistent buffering configuration for incoming envelopes.
 #[derive(Debug, Clone, Serialize, Deserialize)]
-pub struct PersistentBuffer {
-    /// The path to the persistent buffer file.
-    path: PathBuf,
+pub struct EnvelopeSpool {
+    /// The path to the persistent spool file.
+    ///
+    /// If set, this will enable the buffering for incoming envelopes.
+    path: Option<PathBuf>,
     /// Maximum number of connections, which will be maintained by the pool.
-    #[serde(default = "buffer_max_connections")]
+    #[serde(default = "spool_envelopes_max_connections")]
     max_connections: u32,
     /// Minimal number of connections, which will be maintained by the pool.
-    #[serde(default = "buffer_min_connections")]
+    #[serde(default = "spool_envelopes_min_connections")]
     min_connections: u32,
     /// The maximum size of the buffer to keep, in bytes.
     ///
-    /// If not set the befault is 10737418240 bytes or 10 GB.
-    max_disk_size: Option<usize>,
+    /// If not set the befault is 524288000 bytes (500MB).
+    #[serde(default = "spool_envelopes_max_disk_size")]
+    max_disk_size: usize,
     /// The maximum number of envelopes to keep in the memory buffer before spooling them to disk.
```

Comment:
```suggestion
    /// The maximum bytes to keep in the memory buffer before spooling envelopes to disk, in bytes.
    ///
    /// This is a hard upper bound. Internally, this is converted to an envelope count by dividing this number by the maximum envelope size, so in practice, because the avg. envelope size is well below the maximum, we start spooling to disk long before this hard limit is reached.
```

Reply:
also done in [15f57c7](https://github.com/getsentry/relay/pull/2012/commits/15f57c7a6b6d26639f69f7c74c864e2ff441c495)